src/main/java/solution/BacktrackingSolution.java

```diff
@@ -0,0 +1,42 @@
+package solution;
+
+import java.awt.*;
+import java.util.List;
+
+class BacktrackingSolution implements Solution {
+
+    private BacktrackingSolution() {
+    }
+
+    static SudokuBoard solve(SudokuBoard board) {
+        if (board == null) {
+            throw new NullPointerException("Board cannot be null.");
+        }
+        System.out.println("Initial board.");
+        printBoard(board);
+        SudokuBoard result = fillInSudoku(board);
+        System.out.println("Result board.");
+        printBoard(board);
+        return result;
+    }
+
+    private static SudokuBoard fillInSudoku(SudokuBoard board) {
+        List<Point> allVariableCoordinates = board.getAllVariableCoordinates();
+        int iterator = 0;
```

Comment:
The name iteration is semantically connected with `Iterator` instance, so might be a bit misleading. E.g. `index` would be better.